### PR TITLE
EOS-22847: Supply DG to CVG mapping to SSPL

### PR DESCRIPTION
**Pull request by 83bhp**
1. Rename data_devices to data-devices
2. Rename metadata_devices to metadata-devices

Signed-off-by: Pritam Bhavsar <pritam.bhavsar@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: lr-cli/cortx_setup/commands/storage/config.py [hunk 1]
```diff
@@ -138,7 +138,6 @@ def __init__(self):
         self.machine_id = None
         self.enclosure_id = None
         self.mode = None
-        self.cvg_count = -1
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/config.py:141]: Should we set this to `self.cvg_count = 0`

File: srv/components/provisioner/files/cluster_sls.template [hunk 2]
```diff
@@ -94,7 +94,8 @@ cluster:
     storage:
       enclosure: enclosure-{{ (node_num + 1) }}
       cvg:
-        - data_devices:
+        - cvg_name:
+          data_devices:
```
Comment (nitin-seagate) [inline srv/components/provisioner/files/cluster_sls.template:98]: The correct template should be:
cvg:
    - cvg_name
    - data_devices
    - metadata_devices

File: lr-cli/cortx_setup/commands/storage/config.py [hunk 3]
```diff
@@ -523,11 +528,10 @@ def run(self, **kwargs):
                         raise ValueError(
                                 f"Validation for data device {device} failed\n"
                                 "Please provide the correct device")
-            cvg_list.insert(self.cvg_count, {'data_devices': data_devices, 'metadata_devices': metadata_devices})
-            self.cvg_count = self.cvg_count + 1
-            self.update_pillar_and_conf('cvg', self.cvg_count)
+            cvg_list.insert(current_cvg_count, {'cvg_name': cvg_name, 'data_devices': data_devices, 'metadata_devices': metadata_devices})
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/config.py:531]: Is the convention correct here?
`data-devices, metadata-devices`

**Review by nitin-seagate — CHANGES_REQUESTED** [review]
Please verify and update accordingly...

File: lr-cli/cortx_setup/commands/storage/config.py [hunk 4]
```diff
@@ -57,7 +57,7 @@ class StorageEnclosureConfig(Command):
     '''
     $ cortx_setup storage config --name <enclosure-name> --type {RBOD|JBOD|EBOD|virtual}
     $ cortx_setup storage config --controller {gallium|indium|virtual} --mode {primary|secondary}
-    $ cortx_setup storage config --cvg {0|1} --metadata_devices <device list> --data_devices <data_devices>
+    $ cortx_setup storage config --cvg dg_name --metadata-devices <device list> --data-devices <device list>
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/config.py:60]: Change the syntax as follows:
`--cvg <disk-group name>`

File: lr-cli/cortx_setup/commands/storage/config.py [hunk 5]
```diff
@@ -116,17 +116,17 @@ class StorageEnclosureConfig(Command):
...
+            'default': None,
             'optional': True,
-            'help': 'Cylinder Volume Group number.'
+            'help': 'Disk group name'
         },
-        'data_devices': {
+        'data-devices': {
             'type': str,
              'optional': True,
             'help': 'List of data devices (Comma separated) e.g /dev/mapper/mpatha,/dev/mapper/mpathb'
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/config.py:127]: Lets change help message:
`List of data devices mapped to CVG disk-group E.g. /dev/mapper/mpatha,/dev/mapper/mpathb`

File: lr-cli/cortx_setup/commands/storage/config.py [hunk 6]
```diff
@@ -116,17 +116,17 @@ class StorageEnclosureConfig(Command):
...
         },
-        'metadata_devices': {
+        'metadata-devices': {
             'type': str,
             'optional': True,
             'help': 'List of metadata devices (Comma separated) e.g /dev/mapper/mpathf,/dev/mapper/mpathg'
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/config.py:132]: Lets change help message:
`List of metadata devices mapped to CVG disk-group E.g. /dev/mapper/mpathx,/dev/mapper/mpathy`

File: lr-cli/cortx_setup/commands/storage/config.py [hunk 7]
```diff
@@ -116,17 +116,17 @@ class StorageEnclosureConfig(Command):
             'help': 'Port of the controller to connect to'
         },
         'cvg': {
-            'type': int,
-            'default': -1,
+            'type': str,
+            'default': None,
             'optional': True,
-            'help': 'Cylinder Volume Group number.'
+            'help': 'Disk group name'
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/config.py:122]: Lets change help message:
`Cylinder Volume Group Name (Disk-Group) E.g. dgA01`

File: lr-cli/cortx_setup/commands/storage/config.py [hunk 8]
```diff
@@ -192,8 +191,8 @@ def update_pillar_and_conf(self, key, value):
         if(key=='password'):
             value = encrypt_secret(value, "storage_enclosure", self.enclosure_id)
 
-        if key == "cvg":
-            value = str(value)
+        # if key == "cvg":
```
Comment (nitin-seagate) [inline lr-cli/cortx_setup/commands/storage/config.py:194]: Remove this block (if not required)

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

File: lr-cli/cortx_setup/commands/storage/config.py (hunk 7 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/config.py]: Updated

File: lr-cli/cortx_setup/commands/storage/config.py (hunk 5 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/config.py]: Done

File: lr-cli/cortx_setup/commands/storage/config.py (hunk 6 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/config.py]: Done

File: lr-cli/cortx_setup/commands/storage/config.py (hunk 1 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/config.py:141]: Not required anymore, it handled on line no 505

File: lr-cli/cortx_setup/commands/storage/config.py (hunk 8 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/config.py]: Removed

File: lr-cli/cortx_setup/commands/storage/config.py (hunk 3 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/storage/config.py:528]: yes, those are variable names.

File: srv/components/provisioner/files/cluster_sls.template (hunk 2 above)
Comment (83bhp) [reply srv/components/provisioner/files/cluster_sls.template:98]: It's correct, cvg itself is a list of cvgs, so convention is correct, will change if I find it wrong in the testing.

**Review by 83bhp — COMMENTED** [review]
Addressed review comments, please check @nitin-seagate

File: srv/components/system/files/confstore_template.j2 [hunk 9]
```diff
@@ -91,6 +91,7 @@ server_node>{{ machine_id }}>network>data>transport_type=>{{ pillar['cluster'][n
 server_node>{{ machine_id }}>storage>cvg_count=>{{ pillar['cluster'][node]['storage']['cvg'] | length }}
 {%- for i in range(pillar['cluster'][node]['storage']['cvg'] | length) %}
 {%- set outer_loop = loop %}
+{%- set cvg_name = pillar['cluster'][node]['storage']['cvg']['i']['cvg_name'] %}
```
Comment (nitin-seagate) [inline srv/components/system/files/confstore_template.j2:94]: I think we need to fetch CVG name and set to ConfStore something like:
`server_node>{{ machine_id }}>storage>cvg[{{ outer_loop.index - 1 }}]>name=>{{ pillar['cluster'][node]['storage']['cvg'][outer_loop.index - 1]['cvg_name'] }}`

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

File: srv/components/system/files/confstore_template.j2 (hunk 9 above)
Comment (83bhp) [reply srv/components/system/files/confstore_template.j2]: Addressed.

**Review by nitin-seagate — APPROVED** [review]
LGTM

**Pull request merged**